### ISSUE TITLE
[css-nesting][css-selectors] Behavior of & with a default namespace?

### ISSUE THREAD

**Issue by Loirooriol** [w3c/csswg-drafts#9806]
https://drafts.csswg.org/selectors/#type-nmsp

> If a [default namespace](https://drafts.csswg.org/css-namespaces-3/#default-namespace) is declared, [compound selectors](https://drafts.csswg.org/selectors/#compound) without [type selectors](https://drafts.csswg.org/selectors/#type-selector) in them still only match elements in that default namespace.

https://drafts.csswg.org/css-nesting/#nest-selector defines a `&` selector (which is not a type selector).

Then, consider

```css
@namespace url("http://example.com/foo");
svg|a {
  &:hover {}
}
```

`&:hover` has no type selector, so as per Selectors, it won't match a `svg|a:hover`.

But I think it's supposed to match to be consistent with what @tabatkins is saying in https://github.com/w3c/csswg-drafts/issues/5684 and https://github.com/w3c/csswg-drafts/issues/9804

Possibly, Selectors should say something like

> If a [default namespace](https://drafts.csswg.org/css-namespaces-3/#default-namespace) is declared, [compound selectors](https://drafts.csswg.org/selectors/#compound) without [type selectors](https://drafts.csswg.org/selectors/#type-selector) <ins>nor `&` selectors</ins> in them still only match elements in that default namespace.

**Comment by tabatkins** [comment]
I agree, `&` should prevent the default namespace from applying. It falls under the general rule that dbaron describes, as I summarized in <https://github.com/w3c/csswg-drafts/issues/9804#issuecomment-1896309229>. This would have to be handled via explicit spec text, as it wouldn't be caught by my proposed general rule there.
